src/Client/ClientBase.cpp

```diff
@@ -720,6 +721,9 @@ catch (...)
 
 void ClientBase::initLogsOutputStream()
 {
+    if (global_context->getApplicationType() == Context::ApplicationType::EMBEDDED_CLIENT && !server_logs_file.empty())
```

Comment:
Let's introduce separate options for Client (i.e. `ClientOptions`, not settings to distguish it from the settings), that will include such things as `server_logs_file`, that way we will document all such places, and then, we can control what we pass from the embedded client